### PR TITLE
Documentation: Updated index module documentation.

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**

### CODE REVIEW

File: assembly/std/index.ts
Comment (0xSwapFeeder) [inline assembly/std/index.ts:443]: @BatiGencho This function seems to not exist in the env module is this normal ? Or should we remove this function ?

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:224]: I thought about something simpler
```suggestion
 * Generate a log in the Massa blockchain
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:226]: The link is a bit confusing as it point to the massa-as-example repo. I would remove it

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:237]: I would not use abreviation in function description. it can make harder for developers to understand the purpose of the function.

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:240]: We can explain what is the smallest unit in a remark.

see: [unit standard](https://github.com/massalabs/massa-standards/blob/main/units.md)
Or just a link. What do you think @gregLibert ?

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:252]: Same here, we can remove SCE

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:255]: What do you mean by that ?

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:252]:
```suggestion
 * Transfer an amount of coins `from` the specified address `to` a target address.
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:281]:
```suggestion
 * Returns the balance of the specified address
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:283]:
```suggestion
 * @param address - The address for which the balance is retrieved.
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:479]: I would keep unsafe
```suggestion
 * Generates an unsafe random integer.
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:598]: Returns the current thread where the contract is being executed

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:612]:
```suggestion
 * The createEvent is meant to be used with the {@link generateEvent} function.
```

File: assembly/std/index.ts
Comment (Ben-Rey) [inline assembly/std/index.ts:635]:
```suggestion
/**
 * Computes the SHA256 hash of the given `data`.
 *
 * @remarks
 * The SHA256 hash algorithm produces a 32-byte hash, which is returned as a `StaticArray<u8>`.
 *
 * @param data - The data to hash.
 *
 * @returns The SHA256 hash of the `data`, serialized as a `StaticArray<u8>`.
 *
 */
export function sha256(data: StaticArray<u8>): StaticArray<u8> {
  return env.sha256(data);
}
```

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts:249]: Transfering coins from an other address than the current one might be protected ?
Didn't found any infos in massa-sc-runtime.

File: assembly/std/index.ts
Comment (0xSwapFeeder) [inline assembly/std/index.ts:326]: Since the function getKeys and getKeysOf are doing the exact same has this function but with optional parameter in addition:  is this function to be marked deprecated ?

File: assembly/std/index.ts
Comment (0xSwapFeeder) [inline assembly/std/index.ts:471]: Can this function go in the Address class as a static function ?

File: assembly/std/index.ts
Comment (0xSwapFeeder) [inline assembly/std/index.ts:633]: Is this function fully generic ?
Even if so, was it built to answer a specific use case to describe to users ?
Such as computing the hash of a contract ?

File: assembly/std/index.ts
Comment (0xSwapFeeder) [inline assembly/std/index.ts:645]: Can this function migrate to the Address class as a static function ?

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:249]: @AurelienFT How do we know we have the right to use assembly_script_transfer_coins_for. For contract that we own ?

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:471]: For now we can leave it like that.

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:645]: For now we can leave it like that. There is an issue to implement validate_address in Address object

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:326]: We can live with this for now as it could be a clue when reading the smart contract code that we want the actual keys.

File: assembly/std/index.ts
Comment (Ben-Rey) [reply assembly/std/index.ts:84]: Since this is a critical value, I think it's best not to set a default value and leave it to the developer of the smart contract.

Comment (0xSwapFeeder) [reply assembly/std/index.ts:84]: Should I rephrase the description to make sure the user understands the argument is not optional but it can take 0 as value ?

File: assembly/std/index.ts
Comment (AurelienFT) [reply assembly/std/index.ts:439]: No, I think this could be done directly on AS side. We don't need massa info for this and so passing to massa-sc-runtime will cost gas for nothing

File: assembly/std/index.ts
Comment (AurelienFT) [reply assembly/std/index.ts:249]: You can only use this endpoint when you have the write access on the address you are passing as parameter. This is possible at SC address contract creation for example

Side note: SCE balance do no exist anymore they have been merged together since few months there is only one balance now.

File: assembly/std/index.ts
Comment (gregLibert) [reply assembly/std/index.ts:234]: Yes, you can reference the standard in a remark!

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts:442]: Since this issue is meant for documentation, should we open another one to implement this then ?

Comment (Ben-Rey) [reply assembly/std/index.ts:442]: Yes good idea.

File: assembly/std/index.ts
Comment (gregLibert) [inline assembly/std/index.ts:90]: Does it mean that the remote function is executed in the local context  ?

**Review by gregLibert — APPROVED** [review]
One simple question. Nothing blocking the merge.

Good job boys!

File: assembly/std/index.ts
Comment (0xSwapFeeder) [reply assembly/std/index.ts:90]: Yes, you want me to add 'current local context' instead ?

**Pull request merged**
